TSPL.docc/LanguageGuide/Protocols.md

```diff
@@ -823,272 +821,147 @@ Delegation can be used to respond to a particular action,
 or to retrieve data from an external source without needing to know
 the underlying type of that source.
 
-The example below defines two protocols for use with dice-based board games:
+The example below defines a dice game
+and a nested protocol for a delegate
+that tracks the game's progress.
 
 ```swift
-protocol DiceGame {
-    var dice: Dice { get }
-    func play()
-}
-protocol DiceGameDelegate: AnyObject {
-    func gameDidStart(_ game: DiceGame)
-    func game(_ game: DiceGame, didStartNewTurnWithDiceRoll diceRoll: Int)
-    func gameDidEnd(_ game: DiceGame)
-}
-```
+class DiceGame {
+    let sides: Int
+    let generator = LinearCongruentialGenerator()
+    weak var delegate: Delegate?
 
-<!--
-  - test: `protocols`
+    init(sides: Int) {
+        self.sides = sides
+    }
 
-  ```swifttest
-  -> protocol DiceGame {
-        var dice: Dice { get }
-        func play()
-     }
-  -> protocol DiceGameDelegate: AnyObject {
+    func roll() -> Int {
+        return Int(generator.random() * Double(sides)) + 1
+    }
+
+    func play(rounds: Int) {
+        delegate?.gameDidStart(self)
+        for round in 1...rounds {
+            let player1 = roll()
+            let player2 = roll()
+            if player1 == player2 {
+                delegate?.game(self, didEndRound: round, winner: nil)
+            } else if player1 > player2 {
+                delegate?.game(self, didEndRound: round, winner: 1)
+            } else {
+                delegate?.game(self, didEndRound: round, winner: 2)
+            }
+        }
+        delegate?.gameDidEnd(self)
+    }
+
+    protocol Delegate: AnyObject {
         func gameDidStart(_ game: DiceGame)
-        func game(_ game: DiceGame, didStartNewTurnWithDiceRoll diceRoll: Int)
+        func game(_ game: DiceGame, didEndRound round: Int, winner: Int?)
         func gameDidEnd(_ game: DiceGame)
-     }
-  ```
--->
+    }
+}
+```
 
-The `DiceGame` protocol is a protocol that can be adopted
-by any game that involves dice.
+The `DiceGame` class implements a game where
+each player takes a turn rolling dice,
+and the player who rolls the highest number wins the round.
+It uses a linear congruential generator,
+from the example earlier in the chapter,
+to generate random numbers for dice rolls.
+
+The `DiceGame.Delegate` protocol can be adopted
+to track the progress of a dice game.
+Because the `DiceGame.Delegate` protocol
+is always used in the context of a dice game,
+it's nested inside of the `DiceGame` protocol.
```

Comment:
```suggestion
it's nested inside of the `DiceGame` class.
```

Or "type", to avoid repeating the word "class" (it's used in the next sentence)?